src/Tasks/Microsoft.NET.Build.Tasks/ValidateExecutableReferences.cs

```diff
@@ -54,6 +57,11 @@ protected override void ExecuteCore()
                 bool referencedProjectIsSelfContained = MSBuildUtilities.ConvertStringToBool(projectAdditionalProperties["SelfContained"]);
                 bool referencedProjectHadSelfContainedSpecified = MSBuildUtilities.ConvertStringToBool(projectAdditionalProperties["_SelfContainedWasSpecified"]);
 
+                // We can only access TargetFrameworks and NearestTargetFramework to find the referenced project "TargetFramework".
+                // We rely on the nearest one because it will pick the lowest 'most-compatible' tfm for the two projects.
+                // Since 'younger' TFMs are the ones that would error and are generally also what gets picked as 'most-copmaptible' we can use it.
+                var referencedProjectTargetFramework = NuGetFramework.Parse(nearestTargetFramework);
```

Comment:
We should not be parsing the TargetFramework here, that will not work well with aliased TargetFrameworks (see [here](https://github.com/NuGet/Home/issues/5154) and [here](https://github.com/NuGet/Home/pull/12124).

I think what we should instead be doing is getting the index of the NearestTargetFramework from the TargetFrameworks metadata, and then retrieving the value from the same index in the TargetFrameworkMonikers metadata.  You can then parse that moniker with the [FrameworkName](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.versioning.frameworkname.-ctor?view=net-8.0#system-runtime-versioning-frameworkname-ctor(system-string)) class, and check that the `Identifier` is `.NETCoreApp` and the `Version` is less than 8.

Here's a sample of what the data for a multi-targeted referenced project looks like from a binlog:

```
../ReferencedProject/ReferencedProject.csproj
    ReferenceSourceTarget = ProjectReference
    MSBuildSourceProjectFile = c:\git\dotnet-sdk-main\artifacts\tmp\Debug\ReferencedExe---C4AC8564\ReferencedProject\ReferencedProject.csproj
    HasSingleTargetFramework = false
    SetTargetFramework = TargetFramework=net8.0
    ReferenceOutputAssembly = true
    IsVcxOrNativeProj = false
    OutputItemType = 
    TargetPlatformMonikers = None;Windows,Version=7.0
    BuildReference = true
    Platform = AnyCPU
    AdditionalPropertiesFromProject = <AdditionalProjectProperties>
  <TargetFramework Name="net8.0">
    <SelfContained>true</SelfContained>
    <_IsExecutable>true</_IsExecutable>
    <IsRidAgnostic>false</IsRidAgnostic>
    <ShouldBeValidatedAsExecutableReference></ShouldBeValidatedAsExecutableReference>
    <_SelfContainedWasSpecified>true</_SelfContainedWasSpecified>
  </TargetFramework>
  <TargetFramework Name="net8.0-windows">
    <SelfContained>false</SelfContained>
    <_IsExecutable>true</_IsExecutable>
    <IsRidAgnostic>false</IsRidAgnostic>
    <ShouldBeValidatedAsExecutableReference></ShouldBeValidatedAsExecutableReference>
    <_SelfContainedWasSpecified></_SelfContainedWasSpecified>
  </TargetFramework>
</AdditionalProjectProperties>
    TargetFrameworkMonikers = .NETCoreApp,Version=v8.0;.NETCoreApp,Version=v8.0
    TargetFrameworks = net8.0;net8.0-windows
    IsRidAgnostic = false
    MSBuildSourceTargetName = GetTargetFrameworks
    NearestTargetFramework = net8.0
    Platforms = AnyCPU
    Targets = 
    OriginalItemSpec = ../ReferencedProject/ReferencedProject.csproj
```

Reply:
Thanks, I didn't consider this and pushed the change. I must admit I don't really get the difference between this and the TargetFrameworkElement retrived using `UseAttributeForTargetFrameworkInfoPropertyNames` and I didn't leverage that property. Please let me know if we need to use that as I don't really see what it is being used for